File: app/src/main/java/com/wadektech/eventshub/adapter/FriendsEventsAdapter.java
package com.wadektech.eventshub.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import com.wadektech.eventshub.R;
import com.wadektech.eventshub.models.FriendsEvents;

import java.util.List;

public class FriendsEventsAdapter extends RecyclerView.Adapter<FriendsEventsAdapter.ViewHolder> {
    public List<FriendsEvents> friendsEventsList ;
    public Context context;
    public OnSingleFriendsEventItemClicked singleFriendsEventItemClicked ;

    public FriendsEventsAdapter(List<FriendsEvents> friendsEventsList, Context context, OnSingleFriendsEventItemClicked singleFriendsEventItemClicked) {
        this.friendsEventsList = friendsEventsList;
        this.context = context;
        this.singleFriendsEventItemClicked = singleFriendsEventItemClicked;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(context).inflate(R.layout.events_list, parent, false);
        return new ViewHolder(view, singleFriendsEventItemClicked);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        FriendsEvents friendsEvents = friendsEventsList.get(position);

        holder.date.setText(String.valueOf(friendsEvents.getDate()));
        holder.entryFees.setText(String.valueOf(friendsEvents.getEntryFees()));
        holder.eventTitle.setText(friendsEvents.getTitle());
        holder.location.setText(friendsEvents.getLocation());
        holder.shortDescription.setText(friendsEvents.getShortDesc());

    }

    @Override
    public int getItemCount() {
        return friendsEventsList.size();
    }

    public class ViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {
        public OnSingleFriendsEventItemClicked friendsEventItemClicked;
        public TextView eventTitle , shortDescription, date, location , entryFees ;

        public ViewHolder(@NonNull View itemView, OnSingleFriendsEventItemClicked friendsEventItemClicked) {
            super(itemView);
            eventTitle = itemView.findViewById(R.id.tv_event_title);
            shortDescription = itemView.findViewById(R.id.tv_title_desc);
            date = itemView.findViewById(R.id.tv_event_date);
            location = itemView.findViewById(R.id.tv_event_location);
            entryFees = itemView.findViewById(R.id.tv_event_fees);
            this.friendsEventItemClicked = friendsEventItemClicked ;
            itemView.setOnClickListener(this);
        }

        @Override
        public void onClick(View v) {
            friendsEventItemClicked.singleFriendsEventsItemClicked(getAdapterPosition());
        }
    }

    public interface OnSingleFriendsEventItemClicked{
        void singleFriendsEventsItemClicked(int position);
    }
}

File: app/src/main/java/com/wadektech/eventshub/adapter/FriendsEventsAdapter.kt
package com.wadektech.eventshub.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.paging.PagedListAdapter
import androidx.recyclerview.widget.DiffUtil
import androidx.recyclerview.widget.ListAdapter
import androidx.recyclerview.widget.RecyclerView
import com.wadektech.eventshub.R
import com.wadektech.eventshub.databinding.FriendsEventsListBinding
import com.wadektech.eventshub.models.FriendsEvents

class FriendsEventsAdapter( var singleFriendsEventItemClicked: OnSingleFriendsEventItemClicked) :
        PagedListAdapter<FriendsEvents, FriendsEventsAdapter.ViewHolder>(FriendsDiffUtil()) {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        return ViewHolder.from(parent)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val friends = getItem(position)
        if (friends != null){
            holder.bind(friends, singleFriendsEventItemClicked)
        }
    }

    class ViewHolder private constructor(val binding: FriendsEventsListBinding) :
            RecyclerView.ViewHolder(binding.root) {

        fun bind(friendsEvents: FriendsEvents, singleFriendsEventClicked: OnSingleFriendsEventItemClicked) {
            binding.tvEventTitle.text= friendsEvents.title
            binding.tvTitleDesc.text = friendsEvents.shortDesc
            binding.tvEventDate.text = friendsEvents.date.toString()
            binding.tvEventLocation.text = friendsEvents.location
            binding.tvEventFees.text = friendsEvents.entryFees.toString()

            itemView.setOnClickListener {
                singleFriendsEventClicked.singleFriendsEventsItemClicked(adapterPosition)
            }
        }
        companion object {
            fun from(parent: ViewGroup): ViewHolder {
                val layoutInflater = LayoutInflater.from(parent.context)
                val binding = FriendsEventsListBinding.inflate(layoutInflater, parent, false)
                return ViewHolder(binding)
            }
        }
    }

    interface OnSingleFriendsEventItemClicked {
        fun singleFriendsEventsItemClicked(position: Int)
    }

    class FriendsDiffUtil : DiffUtil.ItemCallback<FriendsEvents>(){
        override fun areItemsTheSame(oldItem: FriendsEvents, newItem: FriendsEvents): Boolean {
            return oldItem.id == newItem.id
        }

        override fun areContentsTheSame(oldItem: FriendsEvents, newItem: FriendsEvents): Boolean {
           return  oldItem.id == newItem.id
        }

    }

}